fetch.js

```diff
@@ -1,7 +1,21 @@
 (function(self) {
   'use strict';
 
-  if (self.fetch) {
+  function canAbortFetch() {
+    if (!self.AbortController || !self.AbortSignal) {
+      return false
+    }
+  
+    var abortController = new self.AbortController()
+  
+    var request = new self.Request('http://a', {
+      signal: abortController.signal
+    })
+  
+    return Boolean(request.signal)
+  }
+
+  if (self.fetch && canAbortFetch()) {
```

Comment:
Does this entirely overwrite the native `window.fetch` implementation if fetch isn't abortable?

I don't think that's acceptable. This polyfill should perhaps *wrap* native fetch to provide it with abort functionality, but not overwrite it entirely.

Reply:
Correct, that is the current behavior. The intention here was to ensure that the HTTP requests are _actually_ canceled to reduce the amount of data being sent over the network.